primitives/src/signatures/mod.rs

```diff
@@ -114,6 +139,43 @@ mod tests {
         assert!(S::verify(&parameters, &pk, message, &sig).is_ok());
     }
 
+    pub(crate) fn agg_sign_and_verify<S: AggregateableSignatureSchemes>(
+        messages: &[&[S::MessageUnit]],
+        bad_message: &[S::MessageUnit],
+    ) {
+        let rng = &mut test_rng();
+        let parameters = S::param_gen(Some(rng)).unwrap();
+        let mut sks = vec![];
+        let mut pks = vec![];
+        let mut sigs = vec![];
+        let mut partial_sigs = vec![];
+        let message_for_msig = messages[0];
+        for message in messages.iter() {
+            let (sk, pk) = S::key_gen(&parameters, rng).unwrap();
+            let sig = S::sign(&parameters, &sk, message, rng).unwrap();
+            let partial_sig = S::sign(&parameters, &sk, message_for_msig, rng).unwrap();
+            sks.push(sk);
+            pks.push(pk);
+            sigs.push(sig);
+            partial_sigs.push(partial_sig);
+        }
+        // happy paths
+        let agg_sig = S::aggregate(&parameters, &sigs).unwrap();
+        let multi_sig = S::aggregate(&parameters, &partial_sigs).unwrap();
+        assert!(S::aggregate_verify(&parameters, &pks, messages, &agg_sig).is_ok());
+        assert!(S::mutli_sig_verify(&parameters, &pks, message_for_msig, &multi_sig).is_ok());
```

Comment:
mutli ->  multi

Reply:
ah yes, thanks for catching! done in 90bd4e62fe6da77347006baeb1eb141a62e35817